rfd/0144-client-tools-updates.md

```diff
@@ -0,0 +1,295 @@
+```
+
+#### Self-managed client tools updates
+
+Cluster administrators that want to self-manage client tools updates will be
+able to watch for changes to client tools versions which can then be used to
+trigger other integrations (using MDM software like JamF) to update the
+installed version of client tools on endpoints.
+
+```
+$ tctl autoupdate watch
+{"tools_version": "1.0.0"}
+{"tools_version": "1.0.1"}
+{"tools_version": "2.0.0"}
+
+[...]
+```
+
+##### Cluster configuration
+
+Enrollment of clients in automatic updates will be enforced at the cluster
+level.
+
+The `cluster_maintenance_config` resource will be updated to allow cluster
+administrators to turn client tools automatic updates `on` (default) or `off`.
+A `autoupdate_version` resource will be added to allow cluster administrators
+to manage the version of tools pushed to clients.
+
+> [!NOTE]
+> Client tools configuration is broken into two resources to [prevent
+> updates](https://github.com/gravitational/teleport/blob/master/lib/modules/modules.go#L332-L355)
+> to `autoupdate_verson` on Cloud.
+>
+> While Cloud customers will be able to use `cluster_maintenance_config` to
+> turn client tools automatic updates `off` and self-manage updates, they will
+> not be able to control the version of client tools in `autoupdate_verson`.
+> That will continue to be managed by the Teleport Cloud team.
+
+Both resources can either be updated directly or by using `tctl` helper
+functions.
+
+```yaml
+kind: cluster_maintenance_config
+spec:
+  # tools_auto_update allows turning client tools updates on or off at the
+  # cluster level. Only turn client tools automatic updates off if self-managed
+  # updates are in place.
+  tools_auto_update: on|off
+
+  [...]
+```
+```
+$ tctl autoupdate update --set-tools-auto-update=off
+Automatic updates configuration has been updated.
+```
+
+All Cloud clusters will be opted into `tools_auto_update: on`.
+
+All new self-hosted clusters will be opted into `tools_auto_update: on`.
+Existing self-hosted clusters will be opted into `tools_auto_update: off` to
+not introduce unexpected behavior.
+
+```yaml
+kind: autoupdate_version
+spec:
+  # tools_version is the version of client tools the cluster will advertise.
+  # Can be auto (match the version of the proxy) or an exact semver formatted
+  # version.
+  tools_version: auto|X.Y.Z
+```
+```
+$ tctl autoupdate update --set-tools-version=1.0.1
+Automatic updates configuration has been updated.
+```
+
+For Cloud clusters, `tools_version` will always be `X.Y.Z`, with the version
+controlled by the Cloud team.
+
+For self-hosted clusters, the default will be `tools_version: auto` matching
+the tools version with the proxy version. This way, automatic updates will be
+available to self-hosted customers by default, without any extra effort from
+the cluster administrator.
+
+The above configuration will then be available from the unauthenticated
+endpoint `/v1/webapi/ping` which clients will consult.
```

Comment:
Should we eliminate `/v1/webapi/automaticupgrades` and centralize the properties to `/v1/webapi/ping`?

Reply:
@sclevine You sure we have that endpoint? I am getting 404 for both.
```
$ curl -sk https://localhost:3080/v1/webapi/automaticupgrades 
404 page not found
$ curl -sk https://localhost:3080/v1/webapi/automaticupdates 
404 page not found
```

That being said, I have no problem with us consolidating around `v1/webapi/automaticupdates`.

Reply:
Updated to use `/v1/webapi/automaticupdates` unless anyone has strong objections and thinks this information should be in `/v1/webapi/ping`.

Reply:
It was previously channel-specific, e.g., `/v1/webapi/automaticupgrades/channel/stable/cloud`.

I have no preference, and slightly prefer `/v1/webapi/ping` so that it's easier get all version info in one request.

Just wanted to make sure that this RFD and https://github.com/gravitational/teleport/pull/39217 are consistent. I'd actually prefer to update the other one to use `/v1/webapi/ping` 🙂